soil_params_Gupta2020_2022/utils.jl

```diff
@@ -0,0 +1,61 @@
+
+function regrid(data, (lon, lat), resolution, transform, nlayers)
+    lat_count = Int(floor((90.0 - (-90.0)) / resolution)) + 1 # how many points we want in the model
+    lon_count = Int(floor((180.0 - (-180.0)) / resolution)) + 1 # how many points we want in the model
+    outdata = zeros(Float32, lon_count, lat_count, nlayers);
```

Comment:
Can you explan what this regrid function does?

Reply:
Right now the data is at 1km by 1km resolution, this regrids it to ~2.5x25 degrees. Currently the coarsening function takes the mean of the data within a larger grid cell and applies a transformation afterwards. In the mean, only the values over land are used, so cells which are part ocean and part land are not biased towards unphysical values by the parameters in the file over the ocean. So, a mask is needed too.

As an example, the parameter alpha is given in log(alpha), because it can span orders of magnitude. We take the mean in log space, and then take the 10^ of the mean, to get the coarse value of alpha needed by the simulation. More broadly, how we go from high resolution data to coarse resolution data is nontrivial for many fields. A simple average or transform of an average may work for most, but Im not sure. We will want to experiment with this, I think. so it made sense to me to have a preliminary regridding function which gets us to O(resolution) of the simulation, and then we can get on the simulation points using the regridding in SpaceVaryingInputs.

Otherwise I think we need to ask more of SpaceVaryingInputs (allow for arbitrary "aggregate" functions for points within a coarse grid cell) and account for the mask too.

Reply:
added a doc string